XF.Material/Platforms/Ios/Renderers/MaterialButtonRenderer.cs

```diff
@@ -305,8 +306,34 @@ private void SetupIcon()
 
                 try
                 {
-                    image = UIImage.FromFile(_materialButton.Image.File) ?? UIImage.FromBundle(_materialButton.Image.File);
-
+                    if (_materialButton.Image != null)
+                    {
+                        image = UIImage.FromFile(_materialButton.Image.File) ?? UIImage.FromBundle(_materialButton.Image.File);
+                    }
+                    else if(!(_materialButton.ImageSource?.IsEmpty ?? true))
+                    {
+                        IImageSourceHandler imageSourceHandler = null;
+                        if (_materialButton.ImageSource is UriImageSource)
+                        {
+                            imageSourceHandler = new ImageLoaderSourceHandler();
+                            image = imageSourceHandler.LoadImageAsync(_materialButton.ImageSource).Result;
```

Comment:
Using Result is not a good option. Can you change this to async?

Reply:
Yes.